### DISCUSSION TITLE
Questions About Strange Dependencies (probably inherited)

### DISCUSSION THREAD

**Discussion by EvilSupahFly** [linuxmint/discussions#108] [Questions] [answered]
In the Linux Mint Forums, I have asked several times about different dependency issues I've encountered which seem to be inherited from the Ubuntu base which I believe shouldn't be actual dependencies.

Case in point: the Bluetooth Stack (https://forums.linuxmint.com/viewtopic.php?p=2359792). As a Desktop PC user with no actual Bluetooth hardware, I feel like removing the Bluetooth Stack (that I don't need and am not using) should not prompt for the removal of the whole Networking stack and the entire Cinnamon desktop environment.

Similarly, the `ttf-mscorefonts-installer` package (https://forums.linuxmint.com/viewtopic.php?t=397708) has some strange ties. When you try to install `ttf-mscorefonts-installer` directly from the repo, you are also prompted to install 8 other packages: python3-debconf, python3-distro-info, python3-distupgrade, python3-update-manager, ubuntu-advantage-tools, ubuntu-release-upgrader-core, update-manager-core, and update-notifier-common - none of which are **actually** required for Microsoft's True Type fonts to install and function, the reasoning for which I did NOT find satisfactory for the assorted sources I sought answers from.

For the most part, Linux Mint has been an extremely satisfactory daily driver and I've been using it since the earliest days. However, with each new release, there are things I hope to see fixed which get passed over, whether because of minimal use case, or low priority, and while this is by no means a complete list, I feel like they're solid examples of the larger issue I wish to illustrate - that being the need to more closely examine the underpinnings of it all and clean things up a little better.

I realize this would probably be a momentous undertaking since many of these strange dependencies are probably inherited from Ubuntu, upstream, but it's been said that a journey of a thousand miles begins with a single step, and I was hoping if there was a means in place by which these could be addressed as they are encountered, or if it's just something we need to put up with as long as Canonical insists on certain ties?

**Comment by SantiBurgos1089** [comment]
Well, technically all of what you mentioned earlier is a part of what Mint users "have to put up with" as you said it. Due to having inherited an Ubuntu base, Ubuntu put the majority of the packages, dependencies and things associated; as well as how it works and what you get when installing a package 

The best example I can provide: if you install Firefox then you get the snap version (and this is due to Ubuntu), in certain packages like this, what Mint does is to override them with their own version and requirements (others being Cinnamon, Nemo and so on, and related to its apps) to prevent issues or to have a different behavior. Continuing with the Firefox, Mint replaces Firefox to not install the snap version and install a version provided by themselves instead.

The thing is, if they want to "fix" this situation, then they **must repackage** each and everything of the existing packages, which they are unable to do, and thus, they only repackage certain packages and the rest they leave them as it is

**Comment by billyswong** [comment, marked as answer]
The `ttf-mscorefonts-installer` package in Debian is clear of those excessive Ubuntu dependencies.
https://easylinuxtipsproject.blogspot.com/p/libre-office.html#ID6

Ideally Linux Mint can copy that version to the Mint repository and take priority over the one provided by Ubuntu.

**Reply by EvilSupahFly** [reply]
It seems like a simple enough task, so I don't understand why we just have to put up with it. Mint already prioritizes its own repos where packages may overlap, so why not add the fonts too?

`┐⁠(⁠￣⁠ヘ⁠￣⁠)⁠┌`

**Reply by EvilSupahFly** [reply]
For those who are interested, I have created a stand-alone PPA just for the Debian version of this package, which frees you of the Ubuntu Bloat, which is [located here](https://launchpad.net/~seann-giffin/+archive/ubuntu/msttcorefonts-minimal) based on [my fork](https://github.com/EvilSupahFly/msttcorefonts-minimal) of the **original** upstream Debian package.

Launchpad can be a little terse sometimes, so I included a detailed README with my fork that includes everything I could think of.
